src/github/SyftGithubAction.ts

```diff
@@ -443,10 +450,19 @@ export async function uploadDependencySnapshot(): Promise<void> {
     fs.readFileSync(githubDependencySnapshotFile).toString("utf8")
   ) as DependencySnapshot;
 
+  let correlator = `${workflow}_${job}`;
+  // if running in a matrix build, it is not possible to determine a unique value,
+  // so a user must explicitly specify the artifact-name input, there isn't any
+  // other indicator of being run within a matrix build, so we must use that
+  // here in order to properly correlate dependency snapshots
+  const artifactInput = getArtifactNameInput();
+  if (artifactInput) {
+    correlator += `_${artifactInput}`;
+  }
+
   // Need to add the job and repo details
   snapshot.job = {
-    correlator:
-      core.getInput("dependency-snapshot-correlator") || `${workflow}_${job}`,
+    correlator: core.getInput("dependency-snapshot-correlator") || correlator,
```

Comment:
Is there a reason to hide `core.getInput("artifact-name")` behind a helper, but not `core.getInput("dependency-snapshot-correlator")`?

Reply:
`dependency-snapshot-correlator` should really not ever be used, it's not a supported parameter, it was originally used for testing. I just left it in to not break anyone who may have set it